src/helpers/assoc.js

```diff
@@ -1,21 +1,7 @@
 /** @license ISC License (c) copyright 2017 original and current authors */
 /** @author Ian Hofmann-Hicks (evil) */
 
-const curry = require('../core/curry')
-const isObject = require('../core/isObject')
-const isString = require('../core/isString')
-const object = require('../core/object')
+const setProp = require('./setProp')
 
-// assoc : String -> a -> Object -> Object
-function assoc(key, val, obj) {
-  if(!isString(key)) {
-    throw new TypeError('assoc: String required for first argument')
-  }
-
-  if(!isObject(obj)) {
-    throw new TypeError('assoc: Object required for third argument')
-  }
-  return object.assign({ [key]: val }, obj)
-}
-
-module.exports = curry(assoc)
+module.exports =
```

Comment:
Although i understand that this will create something none pure, do we want to do some console logging here to inform the dev?

Reply:
Well no, I do not think that would make sense as we do not allow for dev vs prod implementations. If we log a warning that means that we are polluting their production space with development messages.

Reply:
yeah fair point